### PR TITLE
change MixItr to be const to avoid newing Crossingframe products

### PR DESCRIPTION
**Pull request by davidlange6**
replaces #17490 after changes suggested by @Dr15Jones

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by dmitrijus** [comment]
+1

File: Validation/EcalDigis/src/EcalDigisValidation.cc [hunk 1]
```diff
@@ -213,26 +213,22 @@ void EcalDigisValidation::analyze(edm::Event const & e, edm::EventSetup const &
...
-    std::auto_ptr<MixCollection<PCaloHit> > 
-      barrelHits( new MixCollection<PCaloHit>( crossingFrame.product() ) );
+    const MixCollection<PCaloHit> barrelHits(crossingFrame.product());
     
     MapType ebSimMap;
-    
-    for (MixCollection<PCaloHit>::MixItr hitItr = barrelHits->begin () ;
-         hitItr != barrelHits->end () ;
-         ++hitItr) {
+    for ( auto const & hitItr : barrelHits ) {
```
Comment (Dr15Jones) [inline Validation/EcalDigis/src/EcalDigisValidation.cc]: The variable name is a bit misleading since it is not an interator anymore.

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
please test

**Pull request merged**